docs/developer/scheduler-concepts.md

```diff
@@ -0,0 +1,145 @@
+- [Actions](#actions)
+- [Plugins](#plugins)
+- [Statements and Transaction Model](#statements-and-transaction-model)
+- [Scenarios](#scenarios)
+- [BindRequests](#bindrequests)
+- [Related Documentation](#related-documentation)
+
+## Overview
+
+KAI Scheduler is built around key concepts that work together to make scheduling decisions. This document explains these concepts for developers working with the scheduler.
+
+The scheduler runs in **cycles**. Each cycle takes a snapshot of the cluster state and makes scheduling decisions through a series of actions.
+
+## The Scheduling Cycle
+
+The scheduler runs in periodic cycles (configurable via `schedulePeriod`). Each cycle follows this flow:
+
+```mermaid
+flowchart LR
+    Start([Cycle Start]) --> Cache[Cache Sync]
+    Cache --> Snapshot[Take Snapshot]
+    Snapshot --> Session[Open Session]
+    Session --> Actions[Execute Actions]
+    Actions --> Close[Close Session]
+    Close --> End([Cycle End])
+    
+    style Start fill:#f5f5f5,stroke:#333
+    style End fill:#f5f5f5,stroke:#333
+    style Snapshot fill:#d4f1f9,stroke:#333
+    style Session fill:#d5f5e3,stroke:#333
+    style Actions fill:#fcf3cf,stroke:#333
+```
+
+1. **Cache Sync**: Ensure all Kubernetes resource informers are up-to-date
+2. **Snapshot**: Capture point-in-time cluster state
+3. **Session**: Create scheduling context with snapshot data
+4. **Actions**: Execute scheduling actions in sequence (Allocate → Consolidate → Reclaim → Preempt → StaleGangEviction)
+   - Each action processes jobs individually, creating and committing/discarding statements per job
+5. **Session Close**: Clean up and prepare for next cycle
+
+## Cache
+
+The **Cache** serves as the authoritative source of cluster state, built from Kubernetes API informers.
+
+### Cache Responsibilities
+
+- **Data Collection**: Aggregate information from multiple API resources
+- **State Maintenance**: Keep track of resource changes over time
+- **Snapshot Generation**: Create consistent point-in-time views
+- **Change Propagation**: Apply committed scheduling decisions back to cluster
+
+## Snapshots
+
+A **Snapshot** captures the cluster state at the start of each scheduling cycle.
+
+Snapshots capture all the cluster resources and state information needed for scheduling decisions, including pods, nodes, queues, pod groups, bind requests, and other relevant Kubernetes objects.
+
+For detailed information about snapshots and the snapshot plugin, see [Snapshot Plugin](../plugins/snapshot.md).
+
+### Why Snapshots Matter
+
+1. **Consistency**: All scheduling decisions in a cycle are based on the same cluster state
+2. **Performance**: Avoids repeated API calls during scheduling
+3. **Debugging**: Provides reproducible state for analysis
+
+## PodGroups
+
+**PodGroups** represent gang scheduling units where multiple pods must be scheduled together as a single atomic operation. They ensure that either all pods in a group are scheduled or none are scheduled.
```

Comment:
Podgroups are not always scheduled atomically - a podgroup can have 10 pods but only require a minMember of 5, meaning that pods 5-10 are scheduled independently. It's more accurate to say that the pod group "defines the gang scheduling requirements" for the workload - all-or-nothing, no gang requirements, or elastic workloads